ocaml/xapi/storage_impl.ml

```diff
@@ -189,16 +186,33 @@ module Vdi = struct
     else
       Vdi_automaton.Detached
 
-  let set_dp_state dp state t =
-    let rest = List.filter (fun (u, _) -> u <> dp) t.dps in
+  (** [get_dp_vm] returns [vm] of type Vm.t. If the mapping between a datapath and vm exists, it will return the vm. Owtherwise a blank vm will be returned*)
+
+  let get_dp_vm dp t =
+    let _vm = List.assoc_opt dp t.dpv in
+    match _vm with
+    | Some vm ->
+        vm
+    | None ->
+        debug
+          "[Vdi.get_dp_vm] No vm was found on the datapath please restart host \
+           to fix" ;
+        vm_of_s ""
+
+  let add_or_update_dp dp vm state t =
+    debug "[Vdi.add_or_update] dp:%s vm:%s" dp (s_of_vm vm) ;
```

Comment:
Can we avoid this debug logline? I'd like to reduce noise in the logs if possible